packages/dev/core/src/Lights/shadowLight.ts

```diff
@@ -396,4 +397,46 @@ export abstract class ShadowLight extends Light implements IShadowLight {
             (this.transformedDirection as any) = null;
         }
     }
+
+    protected _viewMatrix: Matrix = Matrix.Identity();
+    protected _projectionMatrix: Matrix = Matrix.Identity();
+
+    /**
+     * Returns the view matrix.
+     * @param faceIndex The index of the face for which we want to extract the view matrix. Only used for point light types.
+     * @returns The view matrix. Can be null, if a view matrix cannot be defined for the type of light considered (as for a hemispherical light, for example).
+     */
+    public getViewMatrix(faceIndex?: number): Nullable<Matrix> {
+        const lightDirection = TmpVectors.Vector3[0];
+
+        let lightPosition = this.position;
+        if (this.computeTransformedInformation()) {
+            lightPosition = this.transformedPosition;
+        }
+
+        Vector3.NormalizeToRef(this.getShadowDirection(faceIndex), lightDirection);
+        if (Math.abs(Vector3.Dot(lightDirection, Vector3.Up())) === 1.0) {

```

Comment:
Should that be done with Epsilon, or equals is enough?

Reply:
I just copy/pasted what is done in the `ShadowGenerator` class, but I think testing against `1` is fine, as even the slightest deviation is ok.